harborclient/v2/shell.py

```diff
@@ -356,15 +356,18 @@ def do_list(cs, args):
     help="will only print what would have been deleted")
 def do_repository_delete(cs, args):
     """Delete repository"""
-    # list all the repositories to delete
-    repositories = cs.repositories.list(cs.client.project)
-    for repo in repositories:
-        if re.match(args.repository, repo['name']):
-            if args.dryrun:
-                print("Would have deleted : %s" % repo['name'])
-            else:
-                cs.repositories.delete_repository(repo['name'])
-                print("Repository %s deleted" % repo['name'])
+    key = args.repository
+    if cs.repositories.is_name(key):
```

Comment:
This change seems remove batch deletion repositories by regex. In addition, the `dryrun` option will make no effect after this change.